ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java

```diff
@@ -428,6 +439,18 @@ public void setFontStyle(@Nullable String fontStyleString) {
     }
   }
 
+  @ReactProp(name = ViewProps.TEXT_DECORATION_LINE)
+  public void setTextDecorationLine(@Nullable String textDecorationLineString) {
+    for(String textDecorationLineSubString:textDecorationLineString.split(" ")) {
+      if ("underline".equals(textDecorationLineSubString)) {
+        mIsUnderlineTextDecorationSet = true;
```

Comment:
You also need to consider the case when the value gets updated. Now once you set the flag to true it will never change (which is incorrect). You should perhaps reset the flags at the beginning of this setter method